### DISCUSSION TITLE
Haier fridge with screen

### DISCUSSION THREAD

**Discussion by AndreMenchi** [Andre0512/hon#96] [Q&A]
![IMG_4149](https://github.com/Andre0512/hon/assets/82719482/2ba476a9-e824-435d-8b65-2a794125bea4)
Hi everyone. I'm an 18 year old boy who is very passionate about technology and home automation. My family bought a Haier refrigerator with the screen (HAIER HCW9919FSGB). Unfortunately I noticed that there is no integration with Home Assistant that fully supports it, but I think that integrated into Home Assistant this refrigerator could have enormous potential. It would be great if you could help me better integrate the refrigerator. I don't know how possible it is but what I imagined was a possible integration with the cameras in the refrigerator (which can be seen from the home assistant), obviously the possibility of regulating the temperature in the various areas of the refrigerator or the humidity, seeing if the refrigerator is it open or not and the coolest thing would be to integrate the screen into Home Assistant so that the screen can be a Home Assistant hub to control all the other home automation devices in the house.
I hope you can help me.
Thank you in advance.

**Comment by Andre0512** [comment]
Hi, we currently already support a few basic haier refrigerator functions. But this integration is for the hOn app and so can theoretically support everything that is displayed in the hOn app. You can create an issue with the data archive of your device, then I can say more and can integrate more functions that are usable from the hOn app. 
I think you could also have luck with the screen, I think it runs over the same API, because I already discovered many translation strings to this device. But I assume that additional endpoints are used than the ones we normally use and are included in the data archive. You could sniff the traffic with a MITM proxy and try to find out how the communication works :slightly_smiling_face:

**Reply by AndreMenchi** [reply]
Thank you very much for the reply. If many of the example functions I mentioned are already built in that would be great. As mentioned, the best and most useful thing for me would be to use the refrigerator screen as a home assistant hub. Unfortunately, however, I don't know how to observe the traffic to understand how the refrigerator communicates. Could you give me some additional information on how to do it? If I could understand how communication works I would send you everything you need to better integrate this model. I thank you again
